### PR TITLE
for gh-49508 changing Doc for DataFrame.astype

### LINKED ISSUE #49508: DOC: make clear that `DataFrame.astype` supports `Series` input
### Pandas version checks

- [X] I have checked that the issue still exists on the latest versions of the docs on `main` [here](https://pandas.pydata.org/docs/dev/)

### Location of the documentation

https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.astype.html

### Documentation problem

Currently, the parameter specification

> dtype: data type, or dict of column name

does not mention that Series are natively supported.

### Suggested fix for documentation

Something along the lines of 

> dtype: string, data type, Series or Mapping of column name
>
> Use a data type like object (string, numpy.dtype, pandas.ExtensionDtype or Python type) to cast entire pandas object to the same type. Alternatively, use a Mapping such as a dictionary of the form {col: dtype, …}, where col is a column label and dtype is the scalar type to cast one or more of the DataFrame’s columns to column-specific types.

One might also want to add: (https://github.com/pandas-dev/pandas/issues/43837)

> The Mapping is not allowed to contain column names that are present in the DataFrame.

I also noticed that the relevant code

https://github.com/pandas-dev/pandas/blob/91111fd99898d9dcaa6bf6bedb662db4108da6e6/pandas/core/dtypes/inference.py#L266-L295

essentially is a weaker version of `isinstance(obj, collections.abc.Mapping)` that I guess was introduced to also catch `Series`.
I would propose to think about replacing this with `isinstance(obj, Mapping | Series)` when appropriate. 

Related: https://github.com/pandas-dev/pandas-stubs/issues/410

### PR DESCRIPTION
**Pull request by ramvikrams**
added the series in input in the doc of DataFrame.astype

closes #49508

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -6004,11 +6004,12 @@ def astype(
 
         Parameters
         ----------
-        dtype : data type, or dict of column name -> data type
-            Use a numpy.dtype or Python type to cast entire pandas object to
-            the same type. Alternatively, use {col: dtype, ...}, where col is a
-            column label and dtype is a numpy.dtype or Python type to cast one
-            or more of the DataFrame's columns to column-specific types.
+        dtype : string, data type, Series or Mapping of column name -> data type
+            Use a data type like object (string, numpy.dtype, pandas.ExtensionDtype
```
Comment (mroeschke) [inline pandas/core/generic.py:6008]:
```suggestion
            Use a string, numpy.dtype, pandas.ExtensionDtype
```

File: pandas/core/generic.py [hunk 2]
```diff
@@ -6004,11 +6004,12 @@ def astype(
...
+            or Python type) to cast entire pandas object to the same type. Alternatively,
```
Comment (mroeschke) [inline pandas/core/generic.py:6009]:
```suggestion
            or Python type to cast entire pandas object to the same type. Alternatively,
```

File: pandas/core/generic.py [hunk 3]
```diff
@@ -6004,11 +6004,12 @@ def astype(
...
+            use {col: dtype, ...}, where col is a column label and dtype is a
```
Comment (mroeschke) [inline pandas/core/generic.py:6010]:
```suggestion
            use a mapping, e.g. {col: dtype, ...}, where col is a column label and dtype is a
```

**Review by mroeschke — COMMENTED** [review]
Also appears to be some code check failures

**Comment by ramvikrams** [comment]
> Also appears to be some code check failures

Thanks I'll fix them up

**Comment by ramvikrams** [comment]
Done the changes sir

**Comment by ramvikrams** [comment]
Done the changes sir

**Comment by mroeschke** [comment]
Looks to be some validation errors:

```
Error: /home/runner/work/pandas/pandas/pandas/core/generic.py:5999:PR06:pandas.DataFrame.astype:Parameter "dtype" type should use "str" instead of "string"
Error: /home/runner/work/pandas/pandas/pandas/core/generic.py:5999:PR06:pandas.Series.astype:Parameter "dtype" type should use "str" instead of "string"
```

**Comment by ramvikrams** [comment]
> Looks to be some validation errors:
> 
> ```
> Error: /home/runner/work/pandas/pandas/pandas/core/generic.py:5999:PR06:pandas.DataFrame.astype:Parameter "dtype" type should use "str" instead of "string"
> Error: /home/runner/work/pandas/pandas/pandas/core/generic.py:5999:PR06:pandas.Series.astype:Parameter "dtype" type should use "str" instead of "string"
> ```

Sir I didn't understand how can a validation error come from line 5999 when I did not even change the line, seems strange.

**Comment by mroeschke** [comment]
This is a shared docstring so it's used by 2 other functions. I think `string` should be be `str` as mentioned in the message

**Comment by ramvikrams** [comment]
> This is a shared docstring so it's used by 2 other functions. I think `string` should be be `str` as mentioned in the message

thanls sir done the changes

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @ramvikrams

**Comment by ramvikrams** [comment]
> Thanks @ramvikrams

Thank you sir for your help all along

**Pull request merged**